connect/api/src/main/java/org/apache/kafka/connect/source/SourceRecord.java

```diff
@@ -46,25 +46,49 @@
     private final Map<String, ?> sourcePartition;
     private final Map<String, ?> sourceOffset;
 
+    /**
+     * @deprecated use {@link SourceRecord.Builder}
+     */
+    @Deprecated
```

Comment:
@hachikuji 
> constructors over the builder since it makes it easy to see which fields are needed

I think the way to do that is to have required args on the Builder constructor